### PR TITLE
Tracing guide

### PR DESCRIPTION
**Pull request by david-martin**
closes https://github.com/Kuadrant/docs.kuadrant.io/issues/70

A follow up PR will add this to the docs site.

### CODE REVIEW

File: doc/observability/tracing.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+Using a tracing interface like the Jaeger UI or Grafana, you can search for trace information by the trace ID.
+You may get the trace ID from logs, or from a header in a sample request you want to troubleshoot.
+You can also search for recent traces, filtering by the service you want to focus on.
+
+Here is an example trace in the Grafana UI showing the total request time from the gateway (Istio), the time to check the curent rate limit count (and update it) in limitador and the time to check auth in Authorino:
+
+<img src="./grafana_trace.png" alt="Trace in Grafana UI" width="800"/>
+
+In limitador, it is possible to enable request logging with trace IDs to get more information on requests.
```
Comment (adam-cattermole) [inline doc/observability/tracing.md:113]: Perhaps it's worth mentioning how here? This requires the log level to be increased to at least debug, so the verbosity must be set to 3 or higher in the limitador CR:

```sh
kubectl patch -n kuadrant-system limitador/limitador --type=merge -p '{"spec":{"verbosity":3}}'
```

**Review by adam-cattermole — CHANGES_REQUESTED** [review]
I've got a couple suggestions here but mostly looks good

File: doc/observability/tracing.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+spec:
+  limits: {}
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: HTTPRoute
+    name: mygateway
+  tracing:
+    endpoint: rpc://tempo.tempo.svc.cluster.local:4317
+    insecure: true
+```
```
[the remark this answers is not in the document]
Comment (david-martin) [reply doc/observability/tracing.md:87]: Thanks for spotting this.
I knew that was the case, but copied the wrong example resources & pasted in the tracing config

File: doc/observability/tracing.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+- A Kubernetes cluster with Istio and Kuadrant installed.
+- A trace collector (e.g., Jaeger or Tempo) configured to support [OpenTelemetry](https://opentelemetry.io/) (OTel).
+
+## Configuration Steps
+
+### Istio Tracing Configuration
+
+Enable tracing in Istio by using the [Telemetry API](https://istio.io/v1.11/docs/tasks/observability/distributed-tracing/configurability/telemetry-api/).
+Depending on your method for installing Istio, you will need to configure a tracing `extensionProvider` in your MeshConfig, Istio or IstioOperator resource as well.
+Here is an example Telemetry and Istio config to sample 100% of requests, if using the Istio Sail Operator.
```
[the remark this answers is not in the document]
Comment (adam-cattermole) [reply doc/observability/tracing.md:18]: I've had a quick try with this from the main branch of istio-ecosystem (without Kuadrant, just tracing setup) and everything is working great

**Review by adam-cattermole — APPROVED** [review]
Thanks for the updates lgtm!

**Pull request merged**